Formula/kn.rb

```diff
@@ -0,0 +1,37 @@
+class Kn < Formula
+  desc "Command-line interface for managing Knative Serving and Eventing resources"
+  homepage "https://github.com/knative/client"
+  url "https://github.com/knative/client.git",
+      tag:      "v0.24.0",
+      revision: "0bbb3ec9d92665d372d8393ef3036ba7e0c069a4"
+  license "Apache-2.0"
+
+  depends_on "bash" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    ENV["GO111MODULE"] = "on"
+    ENV["TAG"]         = "v#{version}"
+
+    system "./hack/build.sh", "--fast"
+
+    bin.install "kn"
+  end
+
+  test do
+    list_plugins_output = shell_output("#{bin}/kn plugins list")
+    assert_match("No plugins found.", list_plugins_output)
+
+    bash_completion_output = shell_output("#{bin}/kn completion bash")
+    assert_match("# bash completion for kn", bash_completion_output)
+
+    zsh_completion_output = shell_output("#{bin}/kn completion zsh")
+    assert_match("# zsh completion for kn", zsh_completion_output)
+
+    version_output = shell_output("#{bin}/kn version")
```

Comment:
I think a check on the version command is good enough

Reply:
generally brew folks suggest something more than just a version check. I'll remove these completion checks and let's see how it goes.

Reply:
A meaningful check would require connecting to a Knative installation, but this is surely out of scope. I don't see much value on the completion checks as they are really just a single call to the cobra mechanism. You could try to test a `kn service list` and check that there is a connection error ;-) Or a `kn service create --help`.

Another command that could be used and that does not require a cluster, is the offline generation mode:

```
kn service create random --image rhuss/random:1.0 --target .
cat ./default/ksvc/random.yaml
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  creationTimestamp: null
  name: random
  namespace: default
spec:
  template:
    metadata:
      annotations:
        client.knative.dev/user-image: rhuss/random:1.0
      creationTimestamp: null
    spec:
      containers:
      - image: rhuss/random:1.0
        name: ""
        resources: {}
status: {}
```

However, this feature is still marked as experimental and the output format will very likely change, so that this test would need to be adapted.

Reply:
Generally the homebrew folks recommend something more than just a version test. So yes, this will be a good test to get things over the line with them